src/librustc_traits/lowering.rs

```diff
@@ -363,10 +365,60 @@ pub fn program_clauses_for_type_def<'a, 'tcx>(
 }
 
 pub fn program_clauses_for_associated_type_def<'a, 'tcx>(
-    _tcx: TyCtxt<'a, 'tcx, 'tcx>,
-    _item_id: DefId,
+    tcx: TyCtxt<'a, 'tcx, 'tcx>,
+    item_id: DefId,
 ) -> Clauses<'tcx> {
-    unimplemented!()
+    // Rule ProjectionEq-Normalize
+    //
+    // ProjectionEq can succeed by normalizing:
+    // ```
+    // forall<Self, P1..Pn, Pn+1..Pm, U> {
+    //   ProjectionEq(<Self as Trait<P1..Pn>>::AssocType<Pn+1..Pm> = U) :-
+    //       Normalize(<Self as Trait<P1..Pn>>::AssocType<Pn+1..Pm> -> U)
+    // }
+    // ```
+    let item = tcx.associated_item(item_id);
+    debug_assert_eq!(item.kind, ty::AssociatedKind::Type);
+    let trait_id = match item.container {
+        ty::AssociatedItemContainer::TraitContainer(trait_ref) => trait_ref,
+        _ => bug!("not an trait container"),
+    };
+
+    // `Self as Trait<P1..Pn>`
+    let trait_ref = tcx.impl_trait_ref(trait_id).unwrap();
```

Comment:
`impl_trait_ref(def_id)` returns the corresponding trait ref for an `impl` block referenced by `def_id`. Here you are not in an `impl` block but in a trait definition, hence you may just do:
```rust
let trait_ref = ty::TraitRef::identity(tcx, trait_id);
```